<launch>
  
  <!-- Load hand geometry parameters -->  
  <include file="$(find gpd)/launch/hand_geometry.launch">
    <arg name="node" value="classify_grasp_candidates" />
  </include>
  
  <!-- Load classifier parameters -->  
  <include file="$(find gpd)/launch/caffe/classifier_3channels.launch">
    <arg name="node" value="classify_grasp_candidates" />
  </include>
  
	<node name="classify_grasp_candidates" pkg="gpd" type="classify_candidates" output="screen">
    
    <!-- Where the point cloud is stored -->    
<!--
    <param name="cloud_file_name" value="/home/andreas/data/bigbird/3m_high_track_spray_adhesive/clouds/NP1_0.pcd" />
-->
    <param name="cloud_file_name" value="/home/baxter/data/bigbird/3m_high_tack_spray_adhesive/clouds/NP1_0.pcd" />
<!--
    <param name="cloud_file_name" value="/media/andreas/2a9b7d00-f8c3-4849-9ddc-283f5b7c206a/data/object_datasets/bb_onesource/pcd/vo5_tea_therapy_healthful_green_tea_smoothing_shampoo_gt.pcd" />
-->
    
    <!-- Plotting parameters -->
    <param name="plot_normals" value="false" />
    <param name="plot_samples" value="false" />    
    <param name="plot_candidates" value="true" />
    <param name="plot_filtered_grasps" value="false" />
    <param name="plot_valid_grasps" value="true" />
    <param name="plot_clusters" value="false" />
    <param name="plot_selected_grasps" value="true" />
    
    <!-- Preprocessing of point cloud -->
    <param name="voxelize" value="true"/>
    <param name="remove_outliers" value="false"/>
    <rosparam param="workspace"> [-1, 1, -1, 1, -1, 1] </rosparam>
    <rosparam param="camera_position"> [0, 0, 0] </rosparam>

    <!-- General parameters -->
    <param name="num_samples" value="2000" />
    <param name="num_threads" value="1" /> 
    <param name="gaussian_table_size" value="10000000" />    
    
    <!-- Parameters for local grasp candidate search -->
    <param name="nn_radius" value="0.01" />
    <param name="num_orientations" value="8" /> <!-- Number of orientations to consider -->
    
    <!-- Filtering of grasp candidates --> 
    <param name="filter_grasps" value="false" /> <!-- on workspace and robot hand aperture -->
    <rosparam param="workspace_grasps"> [0.6, 1.0, -0.33, 0.13, -0.3, 1.0] </rosparam>
    <param name="filter_half_antipodal" value="true"/> <!-- on half antipodal -->
    
    <!-- Grasp image creation -->
    <param name="create_image_batches" value="false" /> <!-- creates grasp images in batches (less memory usage) -->
        
    <!-- Clustering of grasps -->
    <param name="min_inliers" value="10" /> <!-- min inliers in each cluster; set to 0 to avoid clustering -->
    
    <!-- Grasp selection -->
    <param name="min_score_diff" value="800" />
    <param name="min_aperture" value="0.3" />
    <param name="max_aperture" value="0.7" />
    <param name="num_selected" value="100" />
        
	</node>
  
</launch>
